main.cc

```diff
@@ -446,6 +446,18 @@ class cql_config_updater {
     }
 };
 
+template <typename Func>
+inline auto defer_noexcept(Func&& func) {
```

Comment:
I think you can give a more shutdown-specific name to this, you already have a shutdown specific log message below. Maybe you could also require a name parameter so we are able to log the name of the component causing trouble.